lib/jormungandr/test/integration/Cardano/Pool/MetricsSpec.hs

```diff
@@ -0,0 +1,124 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE LambdaCase #-}
+module Cardano.Pool.MetricsSpec where
+
+import Prelude
+
+import Cardano.BM.Configuration.Static
+    ( defaultConfigStdout )
+import Cardano.BM.Trace
+    ( nullTracer )
+import Cardano.Pool.Metrics
+    ( monitorStakePools )
+import Cardano.Wallet.Jormungandr
+    ( toSPBlock )
+import Cardano.Wallet.Jormungandr.Launch
+    ( withConfig )
+import Cardano.Wallet.Jormungandr.Network
+    ( JormungandrBackend (..), withJormungandr, withNetworkLayer )
+import Cardano.Wallet.Primitive.Types
+    ( BlockHeader (..), SlotId )
+import Control.Concurrent
+    ( forkIO, killThread )
+import Control.Exception
+    ( bracket, throwIO )
+import Control.Monad
+    ( void )
+import Data.List
+    ( isPrefixOf, (\\) )
+import Fmt
+    ( pretty )
+import System.FilePath
+    ( (</>) )
+import System.IO.Temp
+    ( withSystemTempDirectory )
+import Test.Hspec
+    ( Spec, around, expectationFailure, it )
+import Test.Integration.Framework.DSL
+    ( eventually )
+
+import qualified Cardano.Pool.DB as Pool
+import qualified Cardano.Pool.DB.Sqlite as Pool
+
+spec :: Spec
+spec = around setup $ do
+    it "a monitorStakePools thread with db, if killed and restarted, catches up\
+        \ with one that stays running" $ \(nl, referenceDB) ->
+        withDB "db.sqlite" $ \db ->
+            withMonitorStakePoolsThread nl db $ \worker -> do
+                eventually $ do
+                    db `shouldContainSameSlotsAs` referenceDB
+                    shouldNotBeEmpty db
+
+                killThread worker
+
+                eventually $
+                    shouldBeBehindBy 3 db referenceDB
+
+                withMonitorStakePoolsThread nl db $ \_ ->
+                    eventually $ db `shouldContainSameSlotsAs` referenceDB
```

Comment:
The test has now ended up looking a fair bid different from originally.

- I was able to remove `waitNBlocks` by using `eventually $ shouldBeBehindBy`.
- I used `shouldContainSameSlotsAs` with a custom failure-message